apps/yggdrasil/lib/yggdrasil/role_resource.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Yggdrasil.RoleResource do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "roles_resources" do
+    belongs_to :role, Yggdrasil.Role
+    belongs_to :resource, Yggdrasil.Resource
+
+    has_many :role_resource_permissions, Yggdrasil.RoleResourcePermission
```

Comment:
I'm not a fan of having to define the associations in the model like this, and with ecto 2.0 I can eliminate RoleResourcePermissions and this can then be defined like
```elixir
many_to_many :permissions, Yggdrasil.Permissions, joins_through: "roles_resources_permissions"
```
I tested this out with the 2.0.0-beta.1 and works like a charm, however can't use 2.0.0-beta.1 because ja_serializer depends on ecto 1.0. Which kind of sucks.

There is a `has_many :permissions, {"roles_resources_permissions", Yggdrasil.Permissions}` form, but that one assume one to many, and two didn't work like I expected. This is technically a many_to_many between roles_resources_permissions and permissions.